src/compaction_filter.rs

```diff
@@ -41,16 +49,32 @@ extern "C" fn filter(
     key_len: size_t,
     value: *const u8,
     value_len: size_t,
-    _: *mut *mut u8,
-    _: *mut size_t,
+    new_value: *mut *mut u8,
+    new_value_len: *mut size_t,
     value_changed: *mut bool,
 ) -> bool {
     unsafe {
+        *new_value = ptr::null_mut();
+        *new_value_len = 0;
+        *value_changed = false;
+
         let filter = &mut *(filter as *mut CompactionFilterProxy);
         let key = slice::from_raw_parts(key, key_len);
         let value = slice::from_raw_parts(value, value_len);
-        *value_changed = false;
-        filter.filter.filter(level as usize, key, value)
+        let mut new_value_v = Vec::default();
+        let filtered = filter.filter.filter(
+            level as usize,
+            key,
+            value,
+            &mut new_value_v,
+            mem::transmute(&mut *value_changed),
+        );
+        if *value_changed {
+            *new_value = &mut new_value_v[0] as *mut u8;
+            *new_value_len = new_value_v.len();
+            mem::forget(new_value_v);
```

Comment:
It the caller's responsibility to free it? Please add some comments

Reply:
It seems that we do not need this feature supplied by RocksDB?

Reply:
you can `libc::strdup` the string and in C `crocksdb_compactionfilter_t::Filter` delete the pointer.